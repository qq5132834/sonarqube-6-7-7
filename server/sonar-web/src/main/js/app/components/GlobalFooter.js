/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import React from 'react';
import { Link } from 'react-router';
import GlobalFooterSonarCloud from './GlobalFooterSonarCloud';
import GlobalFooterBranding from './GlobalFooterBranding';
import { translate, translateWithParameters } from '../../helpers/l10n';

/*::
type Props = {
  hideLoggedInInfo?: boolean,
  productionDatabase: boolean,
  onSonarCloud?: { value: string },
  sonarqubeVersion?: string
};
*/

export default function GlobalFooter(
  { hideLoggedInInfo, productionDatabase, onSonarCloud, sonarqubeVersion } /*: Props */
) {
  if (onSonarCloud && onSonarCloud.value === 'true') {
    return <GlobalFooterSonarCloud hideLoggedInInfo={hideLoggedInInfo} />;
  }

  return (
    <div id="footer" className="page-footer page-container">
      {productionDatabase === false && (
        <div className="alert alert-danger">
          <p className="big" id="evaluation_warning">
            {translate('footer.production_database_warning')}
          </p>
          <p>{translate('footer.production_database_explanation')}</p>
        </div>
      )}

      <GlobalFooterBranding />

      <div>
        {!hideLoggedInInfo &&
          sonarqubeVersion &&
          translateWithParameters('footer.version_x', sonarqubeVersion)}
        {!hideLoggedInInfo && sonarqubeVersion && ' - '}
        <a href="http://www.gnu.org/licenses/lgpl-3.0.txt">{translate('footer.licence')}</a>
        {' - '}
        <a href="http://www.sonarqube.org">{translate('footer.community')}</a>
        {' - '}
        <a href="https://redirect.sonarsource.com/doc/home.html">
          {translate('footer.documentation')}
        </a>
        {' - '}
        <a href="https://redirect.sonarsource.com/doc/community.html">
          {translate('footer.support')}
        </a>
        {' - '}
        <a href="https://redirect.sonarsource.com/doc/plugin-library.html">
          {translate('footer.plugins')}
        </a>
        {!hideLoggedInInfo && ' - '}
        {!hideLoggedInInfo && <Link to="/web_api">{translate('footer.web_api')}</Link>}
        {!hideLoggedInInfo && ' - '}
        {!hideLoggedInInfo && <Link to="/about">{translate('footer.about')}</Link>}
      </div>
    </div>
  );
}
