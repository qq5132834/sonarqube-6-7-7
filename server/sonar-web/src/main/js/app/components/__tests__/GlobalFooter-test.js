/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import React from 'react';
import GlobalFooter from '../GlobalFooter';

it('should render the only logged in information', () => {
  expect(shallow(<GlobalFooter productionDatabase={true} />)).toMatchSnapshot();
});

it('should not render the only logged in information', () => {
  expect(
    shallow(
      <GlobalFooter
        hideLoggedInInfo={true}
        productionDatabase={true}
        onSonarCloud={{ value: 'false' }}
        sonarqubeVersion="6.4-SNAPSHOT"
      />
    )
  ).toMatchSnapshot();
});

it('should show the db warning message', () => {
  expect(shallow(<GlobalFooter productionDatabase={false} />).find('.alert')).toMatchSnapshot();
});

it('should display the sq version', () => {
  expect(
    shallow(<GlobalFooter productionDatabase={true} sonarqubeVersion="6.4-SNAPSHOT" />)
  ).toMatchSnapshot();
});

it('should render SonarCloud footer', () => {
  expect(
    shallow(<GlobalFooter productionDatabase={true} onSonarCloud={{ value: 'true' }} />)
  ).toMatchSnapshot();
});
