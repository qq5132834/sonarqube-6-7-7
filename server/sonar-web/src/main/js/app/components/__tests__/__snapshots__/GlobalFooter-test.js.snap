// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the sq version 1`] = `
<div
  className="page-footer page-container"
  id="footer"
>
  <GlobalFooterBranding />
  <div>
    footer.version_x.6.4-SNAPSHOT
     - 
    <a
      href="http://www.gnu.org/licenses/lgpl-3.0.txt"
    >
      footer.licence
    </a>
     - 
    <a
      href="http://www.sonarqube.org"
    >
      footer.community
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/home.html"
    >
      footer.documentation
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/community.html"
    >
      footer.support
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/plugin-library.html"
    >
      footer.plugins
    </a>
     - 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/web_api"
    >
      footer.web_api
    </Link>
     - 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/about"
    >
      footer.about
    </Link>
  </div>
</div>
`;

exports[`should not render the only logged in information 1`] = `
<div
  className="page-footer page-container"
  id="footer"
>
  <GlobalFooterBranding />
  <div>
    <a
      href="http://www.gnu.org/licenses/lgpl-3.0.txt"
    >
      footer.licence
    </a>
     - 
    <a
      href="http://www.sonarqube.org"
    >
      footer.community
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/home.html"
    >
      footer.documentation
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/community.html"
    >
      footer.support
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/plugin-library.html"
    >
      footer.plugins
    </a>
  </div>
</div>
`;

exports[`should render SonarCloud footer 1`] = `<GlobalFooterSonarCloud />`;

exports[`should render the only logged in information 1`] = `
<div
  className="page-footer page-container"
  id="footer"
>
  <GlobalFooterBranding />
  <div>
    <a
      href="http://www.gnu.org/licenses/lgpl-3.0.txt"
    >
      footer.licence
    </a>
     - 
    <a
      href="http://www.sonarqube.org"
    >
      footer.community
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/home.html"
    >
      footer.documentation
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/community.html"
    >
      footer.support
    </a>
     - 
    <a
      href="https://redirect.sonarsource.com/doc/plugin-library.html"
    >
      footer.plugins
    </a>
     - 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/web_api"
    >
      footer.web_api
    </Link>
     - 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/about"
    >
      footer.about
    </Link>
  </div>
</div>
`;

exports[`should show the db warning message 1`] = `
<div
  className="alert alert-danger"
>
  <p
    className="big"
    id="evaluation_warning"
  >
    footer.production_database_warning
  </p>
  <p>
    footer.production_database_explanation
  </p>
</div>
`;
